test/jobs/analyze_change_job_test.rb

```diff
@@ -0,0 +1,31 @@
+require 'test_helper'
+require 'minitest/mock'
+
+class AnalyzeChangeJobTest < ActiveJob::TestCase
+  test 'produces change with annotations' do
+    change = changes(:page1_change_1_2)
+
+    differ_mock = Minitest::Mock.new(Differ)
+    def differ_mock.diff(_change)
+      {
+        'diff' => [
+          [1, 'addition'],
+          [-1, 'subtraction']
+        ]
+      }
+    end
```

Comment:
Yep, this is a good mock. There’s more stuff in the envelope object that wraps the diff, but it’s generally not relevant to this job. But if you’re curious, a more complete example would be:

```js
{
  "version": "1.2.3+hash",
  "type": "links",
  "change_count": 2,
  // other fields specific to selected diff type...

  // Diff is usually this format, but some diffs return something else, like
  // an HTML string (e.g. html_token). All the ones relevant for this test
  // follow this format, though. :)
  "diff": [
    [1, 'addition'],
    [-1, 'subtraction']
  ]
}
```

Reply:
Thanks! I updated the hash to match a little better in case I have this question again in the future.